wagtail/core/blocks/stream_block.py

```diff
@@ -543,6 +543,31 @@ def get_prep_value(self):
 
         return prep_value
 
+    def blocks_by_type(self):
+        blocks = {}
```

Comment:
Taking another look at this... I think it makes sense for the return value here to include an empty list for any block type that is part of the definition but has no blocks. That also means that we can simplify the code below...

    def blocks_by_type(self):
        blocks = {block.name: [] for block in self.stream_block.child_blocks}
        for item in self:
            blocks[item.block_type] = item
        return blocks